awx/main/tasks.py

```diff
@@ -1160,7 +1114,7 @@ def build_passwords(self, job, **kwargs):
         Build a dictionary of passwords for SSH private key, SSH user, sudo/su
         and ansible-vault.
         '''
-        passwords = super(RunJob, self).build_passwords(job, **kwargs)
+        passwords = BaseTask.build_passwords(self, job, **kwargs)
```

Comment:
Had to change this because of the way I implemented the `@task` decorator; making it work the same w/ `super()` required some pretty gross metaclassing nastiness that wasn't easy to understand (it turns out celery does something similar), so I just went with this.

Reply:
Could you leave a comment (or many comments) to this effect? I see this as trivial to work around from a developer perspective, but there are no guardrails to prevent someone from doing this, which would otherwise be ingrained in muscle memory and best practice.